### ISSUE TITLE
Enhancement Request: PolyLine onPress event should return the coordinate of where it was clicked

### ISSUE THREAD

**Issue by tomgiam** [react-native-maps/react-native-maps#4535]
### Summary

Can the PolyLine onPress event be enhanced to return the coordinate of where in the PolyLine it was clicked?
Currently, the PolyLine onPress event always returns the coordinate and map position of the first point in the PolyLine

### Reproducible sample code

```TSX
handlepolylinePress = async (polyline, e) => {
    console.log(e.nativeEvent);
  };

                <Polyline
                  tracksViewChanges={false}
                  tappable={true}
                  key={index}
                  onPress={(event) => this.handlepolylinePress(polyline, event)}
                  coordinates={polyline.coordinates}
                  strokeColor={"rgba(255, 0, 0, 0.30)"}
                  lineDashPattern={[50, 10]}
                  strokeWidth={5}
                />
```

### Steps to reproduce

Map with Polyline and onPress event

### Expected result

Coordinate of where PolyLne was clicked

### Actual result

Coordinate of first point in PolyLine.

### React Native Maps Version

0.31.1

### What platforms are you seeing the problem on?

Android, iOS (Google Maps)

### React Native Version

0.69.6

### What version of Expo are you using?

SDK 46

### Device(s)

Samsung S20+, iPhone 6

**Comment by rodrigodiasf1984** [comment]
I've tried and the console.log return is undefined. 

``` <Polyline
                coordinates={travelState.nearEncounterPoints}
                strokeColor={Colors.rojoUbicacion}
                strokeWidth={9}
                onPress={e =>
                  console.log(
                    'Polyline',
                    e.nativeEvent.coordinate,
                    e.nativeEvent.position,
                  )
                }
                tappable={true}
              />````

Do you have any idea how to solve it? Thx.

**Comment by tomgiam** [comment]
> I've tried and the console.log return is undefined.
> 
> ```
>                 coordinates={travelState.nearEncounterPoints}
>                 strokeColor={Colors.rojoUbicacion}
>                 strokeWidth={9}
>                 onPress={e =>
>                   console.log(
>                     'Polyline',
>                     e.nativeEvent.coordinate,
>                     e.nativeEvent.position,
>                   )
>                 }
>                 tappable={true}
>               />````
> 
> Do you have any idea how to solve it? Thx.
> ```

I do this and it "works", but it always returns the point or coordinate to the first location in the polyline.
I tried position to see if it worked any differently than coordinate, but the same will happen for e.nativeEvent.coordinate

```
onPress={(e) => this.handlepolylinePress(e)}
...
handlepolylinePress = async (e) => {
    e.stopPropagation();
    let loc = await this.mapRef.coordinateForPoint(e.nativeEvent.position);
    console.log("handlepolylinePress", loc);
  };
```

**Comment by rodrigodiasf1984** [comment]
> > I've tried and the console.log return is undefined.
> > ```
> >                 coordinates={travelState.nearEncounterPoints}
> >                 strokeColor={Colors.rojoUbicacion}
> >                 strokeWidth={9}
> >                 onPress={e =>
> >                   console.log(
> >                     'Polyline',
> >                     e.nativeEvent.coordinate,
> >                     e.nativeEvent.position,
> >                   )
> >                 }
> >                 tappable={true}
> >               />````
> > 
> > Do you have any idea how to solve it? Thx.
> > ```
> 
> I do this and it "works", but it always returns the point or coordinate to the first location in the polyline. I tried position to see if it worked any differently than coordinate, but the same will happen for e.nativeEvent.coordinate
> 
> ```
> onPress={(e) => this.handlepolylinePress(e)}
> ...
> handlepolylinePress = async (e) => {
>     e.stopPropagation();
>     let loc = await this.mapRef.coordinateForPoint(e.nativeEvent.position);
>     console.log("handlepolylinePress", loc);
>   };
> ```

Yes, I do have the same problem, the coordinates returned are different from the polyline.

**Comment by tomgiam** [comment]
My issue is slightly different. When I click anywhere on the polyline it always returns the first point of the polyline.

**Comment by Simon-TechForm** [comment]
Hello 👋

This was fixed/changed in [v1.0.0](https://github.com/react-native-maps/react-native-maps/releases/tag/v1.0.0) 👍
